### DISCUSSION TITLE
[Documentation / History of clapper] Perhaps an entry to the main README or a link to the history

### DISCUSSION THREAD

**Discussion by rubyFeedback** [Rafostar/clapper#142] [General]
This is a request to add a bit history for clapper. I assume you guys did not start from zero
but had some knowledge already, be it design-related or multimedia-related.

Back when mpv was new-ish, I asked the mpv devs what the difference was to mplayer
and how it was created. They actually added an explanation - that explanation can still
be found on their FAQ. So my question helped add more information! :)

(Although I use mpv most of the time, I still like mplayer too, even if it is now the
dinosaur guy ...)

So perhaps clapper originated from some other project or other ideas. I'd be interested
to hear how it started and what it aims to achieve. I am sure there is some interesting
background story.

This does not need to be in the main README per se; some projects want to keep
the main README clean and to the point, which I can understand. But if someone can
write some paragraph about the background, and then link to it from the main README,
this would be super-interesting - at the least for me, but perhaps also for other folks
who never heard about clapper before. I also normally report asking for more screenshots
but you folks already have plenty screenshots, so there is no point in asking for even
more screenshots. ;)

(Though it would be interesting to hear if this works on windows too but this is a secondary
question; the primary one is the background story of clapper.)

**Comment by Rafostar** [comment]
This isn't an issue or feature request per-se, so I will move this into github discussions panel before answering.

**Comment by Rafostar** [comment]
> I assume you guys did not start from zero but had some knowledge already, be it design-related or multimedia-related.

Clapper is made by one person entirely, hence the development is not super fast but it goes steady. I do it mainly as a hobby in my free time, and share what I made with the world. I had some basic knowledge on how to use GStreamer (which is used here as a video playback backend), had some ideas whats needed to make it work together with GTK4 and that's basically it. Also some basic multimedia-related knowledge. As for making designs not so much, I just do it how I like it :smile: 

It basically started from [this reddit post](https://www.reddit.com/r/gnome/comments/io6ixj/what_do_you_expect_from_gnome_media_player_what/), where I explained my motivation in trying to make a GNOME targeting video player. Post can still be read today for more details, but I guess the main trigger for me was a MPV commit that caused it to forcefully exit itself when it detected GNOME desktop (reverted some time afterwards due to people protests).

> I also normally report asking for more screenshots but you folks already have plenty screenshots [...] Though it would be interesting to hear if this works on windows too but this is a secondary question;

In its early days the main desktops this project targets are GNOME and Phosh. You can see how the player presents itself on GNOME in screenshot from Readme. There are some videos on YouTube showing how it works on Phosh desktop on PinePhone like [this one here](https://www.youtube.com/watch?v=BvmRV6IIGGo).

Even through project targets mainly GNOME at the moment, it can be used on other desktops environments. For this Flatpak packages are recommended. As for Windows, then maybe in a long distant future. Only if it gains more attention/popularity and if next GStreamer and GTK versions can improve their Vulkan based video decoding and rendering.
